SwrveSDK/SwrveUtils.brs

```diff
@@ -187,78 +293,123 @@ function SwrveCopy(obj as Object) as Object
 	return res
 end function
 
-'Util function depnding on resolution, return supported width and height'
-function SWGetSupportedResolution()
+'Util function for checking to see if the file(or folder) exists'
+function SWCheckForFile(path as String) as Boolean
+	pathParts = SWSplitFileFromPath(path)
+	items = ListDir(pathParts[0]).toArray()
+	for each item in items
+		if item = pathParts[1] OR item = pathParts[1] + ".tmp" then
+			return true
+		end if
+	end for
+	return false
+end function
 
-	supportedWidth =  SwrveConstants().SWRVE_FHD_WIDTH
-	supportedHeight = SwrveConstants().SWRVE_FHD_HEIGHT
+'Util function for getting the last folder or file nice from a uri path.'
+'Returns an array with index 0 is the path before the file/folder and index 1 is the file or folder name'
+function SWSplitFileFromPath(path as String) as Object
+	pathParts = path.tokenize("/").toArray()
+	parts = ["", ""]
 
-	appInfo = CreateObject("roAppInfo")
-    ui_resolutions = appInfo.GetValue("ui_resolutions").trim()
-	if LCase(ui_resolutions) <> "fhd"
-		di = CreateObject("roDeviceInfo")
-		uiRes = di.GetDisplaySize() 
-		supportedWidth =  uiRes.w
-		supportedHeight = uiRes.h
+	if pathParts.count() >= 2 then
+		parts[1] = pathParts.pop()
+		parts[0] = pathParts.join("/") + "/"
 	end if
 
-	res = {}
-	res["width"] = supportedWidth
-	res["height"] = supportedHeight
-	return res
+	return parts
+end function
+
+'Returns 'value' if set or 'defaultValue' if not set.
+function SWGetValue(value as Dynamic, defaultValue as Dynamic) as Dynamic
+	if value <> Invalid then return value else return defaultValue
+end function
+
+'Util function depending on resolution, return supported width and height'
+function SWGetSupportedResolution()
+	return m.top.getScene().currentDesignResolution
```

Comment:
The complicated logic of looking at the manifest and other values was replaced with a simple `currentDesignResolution` on the scene node:
https://developer.roku.com/en-ca/docs/references/scenegraph/abstract-nodes/scene.md

Reply:
Nice 👍